src/projections/gtoc.cpp

```diff
@@ -0,0 +1,264 @@
+    double az = lp.lam;
+    double el = ell2sph(Q->epr2, lp.phi);
+
+    // Convert Spherical to Cartesian:
+    vec v_in = sph2cart(rho, az, el);
+
+    // Perform Rotation:
+    vec v_out = mult(Q->e2m, v_in);
+
+    // Convert Cartesian to Spherical
+    vec rhoAzEl = cart2sph(v_out.x, v_out.y, v_out.z);
+
+    // Apply Scale and Offset factors
+    az = rhoAzEl.y;
+    el = rhoAzEl.z;
+    xy.x = -Q->scale * (el + Q->theta0);
+    xy.y = Q->scale * cos(Q->theta0) * az;
+
+    return xy;
+}
+
+static PJ_LP gtoc_e_inverse(PJ_XY xy, PJ *P) { /* Ellipsoidal, inverse */
+    PJ_LP lp = {0.0, 0.0};
+    struct pj_opaque *Q = static_cast<struct pj_opaque *>(P->opaque);
+
+    // Conversion to spherical coordinates:
+    double rho = Q->scale;
+    double az = xy.y / (cos(Q->theta0) * Q->scale);
+    double el = -xy.x / Q->scale - Q->theta0;
+
+    // Conversion to Cartesian:
+    vec v_in = sph2cart(rho, az, el);
+
+    // Perform Rotation
+    vec v_out = mult(Q->m2e, v_in);
+
+    // Conversion to Spherical Coordinates
+    vec rhoAzEl = cart2sph(v_out.x, v_out.y, v_out.z);
+
+    lp.lam = rhoAzEl.y;
+    lp.phi = sph2ell(Q->epr2, rhoAzEl.z);
+
+    return lp;
+}
+
+PJ *PROJECTION(gtoc) {
+    struct pj_opaque *Q =
+        static_cast<struct pj_opaque *>(calloc(1, sizeof(struct pj_opaque)));
+    if (nullptr == Q)
+        return pj_default_destructor(P, ENOMEM);
+    P->opaque = Q;
+
+    // Earth ellipsoid parameters:
+    double equR = 6378137.0;
+    double polR = 6356752.3;
+    double equR2 = pow(equR, 2);
+    double polR2 = pow(polR, 2);
+
+    // Initializing projection parameters
+    // Note: calloc call above will init all values to 0.0 by default.
+    Q->lon0 = P->lam0;
+    Q->lat0 = P->phi0;
+
+    // Reset the projection parameters:
+    P->lam0 = 0.0;
+    P->phi0 = 0.0;
+
+    if (pj_param(P->ctx, P->params, "taz_0").i) {
+        Q->az0 = pj_param(P->ctx, P->params, "raz_0").f;
+    }
+
+    if (pj_param(P->ctx, P->params, "ttheta_0").i) {
+        Q->theta0 = pj_param(P->ctx, P->params, "rtheta_0").f;
+    }
+
+    if (pj_param(P->ctx, P->params, "tscale").i) {
+        Q->scale = pj_param(P->ctx, P->params, "dscale").f;
+    } else {
+        // No scale value explicitly provided so we compute the default scale
+        // here:
+        double CLat = cos(Q->lat0);
+        double Clat2 = pow(CLat, 2);
+        double SLat2 = 1. - Clat2;
+        Q->scale = sqrt((pow(equR2, 2) * Clat2 + pow(polR2, 2) * SLat2) /
+                        (equR2 * Clat2 + polR2 * SLat2));
+    }
+
+    Q->epr2 = equR2 / polR2;
+
+    // Initializing the trnansformation matrices:
+    // Converting Geotedic to Spheric
+    mat tmp1 = matmult(rotY(Q->theta0), rotX(-Q->az0));
+    mat tmp2 = matmult(rotY(ell2sph(Q->epr2, Q->lat0)), rotZ(-Q->lon0));
+    Q->e2m = matmult(tmp1, tmp2);
+
+    Q->m2e = transpose(Q->e2m);
+
+    // Prevent the use of CLASSIC units by default for outputs:
+    P->right = PJ_IO_UNITS_WHATEVER;
```

Comment:
actually is there a good reason to do so ? Reviewing my above suggestions of using P->a / P->b, why not using the unit sphere/ellipsoid with the flattening of the ellipsoid considered ?